provider/azure/environ.go

```diff
@@ -1700,6 +1805,71 @@ func isControllerDeployment(deployment resources.DeploymentExtended) bool {
 	return false
 }
 
+// allProvisionedInstances returns all of the instances
+// in the given resource group.
+func (env *azureEnviron) allProvisionedInstances(
+	ctx context.ProviderCallContext,
+	resourceGroup string,
+	controllerUUID string,
```

Comment:
if "controllerUUID" != "", we only look for controller instances rather than instances managed by the current controller?

Reply:
yes, we look for controller instances with that controller uuid